packages/ui-components/src/lib/providers/filters/RaindexFilterStore.ts

```
@@ -0,0 +1,114 @@
+ * a clean fluent API for updating filters with automatic persistence.
+ *
+ * This wrapper solves the TypeScript typing issues with WASM generics
+ * by providing a pure TypeScript builder pattern interface.
+ */
+export class RaindexFilterStore {
+	private wasmStore: RaindexFilterStoreWasm;
+
+	constructor() {
+		const wasmStoreResult = RaindexFilterStoreWasm.create();
+		this.wasmStore = unwrapWasmResult(wasmStoreResult);
+		return this;
```

Biome (2.1.2)

[error] The constructor should not return a value.

The constructor is here:

Returning a value from a constructor may confuse users of the class.

(lint/correctness/noConstructorReturn)